src/app.module.ts

```diff
@@ -126,6 +129,21 @@ export class AppModule implements NestModule {
           // return 500 for files that do not exist instead of a 404
           exclude: ['/(.*)'],
         }),
+        TypeOrmModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: async (configService: ConfigService) => {
+            const typeormConfig = await configService.getOrThrow('typeorm');
+            const postgresConfigObject = postgresConfig(
+              await configService.getOrThrow('db.postgres'),
+            );
+
+            return {
+              ...typeormConfig,
+              ...postgresConfigObject,
+            };
```

Comment:
Will this throw if the respective configs don't match the expected structure? Reason being, `getOrThrow` returns a "typeless" object and we have no warning until runtime. This is generally why we `getOrThrow` primitives.

Reply:
No, this does not throw in case types are different. On the other hand, we have configuration tests.

Reply:
Good point. In general, any issues would cause a crash before potential issues occur, e.g. migration breaks.